app/assets/javascripts/preferences/panes/listed/BlogItem.tsx

```diff
@@ -0,0 +1,79 @@
+import { Button } from '@/components/Button';
+import { HorizontalSeparator } from '@/components/shared/HorizontalSeparator';
+import { LinkButton, Subtitle } from '@/preferences/components';
+import { WebApplication } from '@/ui_models/application';
+import { Action, SNComponent, SNItem } from '@standardnotes/snjs/dist/@types';
+import { FunctionalComponent } from 'preact';
+import { useState } from 'preact/hooks';
+
+type Props = {
+  item: SNComponent;
+  showSeparator: boolean;
+  disabled: boolean;
+  disconnect: (item: SNItem) => void;
+  application: WebApplication;
+};
+
+export const BlogItem: FunctionalComponent<Props> = ({
+  item,
+  showSeparator,
+  disabled,
+  disconnect,
+  application,
+}) => {
+  const applicationAlertService = application.alertService;
+
+  const [isDisconnecting, setIsDisconnecting] = useState(false);
+
+  const handleDisconnect = () => {
+    setIsDisconnecting(true);
+    applicationAlertService
+      .confirm(
+        'Disconnecting will result in loss of access to your blog. Ensure your Listed author key is backed up before uninstalling.',
+        `Disconnect blog "${item.name}"?`,
+        'Disconnect',
+        1
+      )
+      .then((shouldDisconnect) => {
+        if (shouldDisconnect) {
+          disconnect(item);
+        } else {
+          setIsDisconnecting(false);
```

Comment:
`setIsDisconnecting(false)` would need to go in a `finally` block to ensure it executes even if there's an error

Reply:
Putting it in the `finally` block turns it off even before it finishes disconnecting, since the `deleteItem()` function is async. I've just put `setIsDisconnecting(false)` in the `catch` block to execute it even if there is an error.

Reply:
Shouldn't we await `disconnect` in that case? Might be a good idea to refactor these to use `async/await` which is a bit easier to read I think

Reply:
That would require the disconnect function to be promisified. That is actually how I had originally implemented it, but I thought I was overcomplicating it, so I removed the promise.

Reply:
@antsgar So I've turned the disconnect function into a Promise and using `await disconnect()` in the BlogItem and `setIsDisconnecting()` inside the `.finally()` block.